Libraries/Opc.Ua.Client/DataDictionary.cs

```diff
@@ -313,27 +314,53 @@ public byte[] ReadDictionary(NodeId dictionaryId)
             // read value.
             DataValueCollection values;
             DiagnosticInfoCollection diagnosticInfos;
+            try
+            {
+                ResponseHeader responseHeader = m_session.Read(
+                    null,
+                    0,
+                    TimestampsToReturn.Neither,
+                    itemsToRead,
+                    out values,
+                    out diagnosticInfos);
 
-            ResponseHeader responseHeader = m_session.Read(
-                null,
-                0,
-                TimestampsToReturn.Neither,
-                itemsToRead,
-                out values,
-                out diagnosticInfos);
+                ClientBase.ValidateResponse(values, itemsToRead);
+                ClientBase.ValidateDiagnosticInfos(diagnosticInfos, itemsToRead);
 
-            ClientBase.ValidateResponse(values, itemsToRead);
-            ClientBase.ValidateDiagnosticInfos(diagnosticInfos, itemsToRead);
+                // check for error.
+                if (StatusCode.IsBad(values[0].StatusCode))
+                {
+                    ServiceResult result = ClientBase.GetResult(values[0].StatusCode, 0, diagnosticInfos, responseHeader);
+                    throw new ServiceResultException(result);
+                }
 
-            // check for error.
-            if (StatusCode.IsBad(values[0].StatusCode))
+                // return as a byte array.
+                return values[0].Value as byte[];
+                
+            }
+            catch (ServiceResultException ex)
             {
-                ServiceResult result = ClientBase.GetResult(values[0].StatusCode, 0, diagnosticInfos, responseHeader);
-                throw new ServiceResultException(result);
+                if (ex.StatusCode != StatusCodes.BadEncodingLimitsExceeded)
+                {
+                    throw;
+                }
+                else
+                {
+                    try
+                    {
+                        byte[] dictionary = m_session.ReadByteStringInChunks(dictionaryId);
+                        return dictionary;
+                    }
+                    catch
+                    {
+                        ExceptionDispatchInfo.Capture(ex).Throw();
```

Comment:
what is this dispatchInfo for?

Reply:
The ReadByteStringInChunks method is called from within a catch block. If it also fails with an exception the original exception is rethrown together with it's call stack information (which, to my knowledge, would be lost if I just throw it).

Reply:
Imho this is not needed when immediately rethrowing like in this case

Reply:
I'm not rethrowing immediately. I'm rethrowing after executing ReadByteStringInChunks.

Reply:
Is the second throw; then ever executed?

Reply:
Now confused. Does this become an aggregatedException?